### PR TITLE
Small fixes and cleanup in SimG4CMS

### PR DESCRIPTION
**Pull request by perrotta**
#### PR description:

Just a few cleanups in the code, mostly to shut up the Static Analizer and simplify the structure

#### PR validation:

It builds
No changes expected

### CODE REVIEW

File: SimG4CMS/Forward/src/TotemRPOrganization.cc [hunk 1]
```diff
@@ -35,15 +35,15 @@ uint32_t TotemRPOrganization ::getUnitID(const G4Step* aStep) const {
   G4VPhysicalVolume* physVol;
   int32_t UNITA = 0;
   const G4VTouchable* touch = aStep->GetPreStepPoint()->GetTouchable();
-  int ii = 0;
-  for (ii = 0; ii < touch->GetHistoryDepth(); ii++) {
-    physVol = touch->GetVolume(ii);
 
 #ifdef SCRIVI
+  for (int ii = 0; ii < touch->GetHistoryDepth(); ii++) {
+    physVol = touch->GetVolume(ii);
```
Comment (perrotta) [inline SimG4CMS/Forward/src/TotemRPOrganization.cc:41]: Hopefully this call has no other purpose than returning the physics volume: correct @civanch ?

**Comment by perrotta** [comment]
please test

File: SimG4Core/CustomPhysics/src/G4muDarkBremsstrahlungModel.cc [hunk 2]
```diff
@@ -285,7 +279,7 @@ G4double G4muDarkBremsstrahlungModel::ComputeCrossSectionPerAtom(
   if (cross < 0.) {
     cross = 0.;
   }
-  E0 = E0 * CLHEP::GeV;
+  //  E0 = E0 * CLHEP::GeV;
```
Comment (civanch) [inline SimG4Core/CustomPhysics/src/G4muDarkBremsstrahlungModel.cc:282]: @perrotta , this line may be removed.

File: SimG4CMS/Forward/src/TotemRPOrganization.cc (hunk 1 above)
Comment (civanch) [reply SimG4CMS/Forward/src/TotemRPOrganization.cc:41]: @perrotta , yes, it is.

File: SimG4Core/CustomPhysics/src/G4muDarkBremsstrahlungModel.cc [hunk 3]
```diff
@@ -265,11 +263,7 @@ G4double G4muDarkBremsstrahlungModel::ComputeCrossSectionPerAtom(
   G.function = &DsigmaDx;
   G.params = &alpha;
   G4double xmin = 0;
-  G4double xmax = 1;
-  if ((muonMass / E0) > (MA / E0))
-    xmax = 1 - muonMass / E0;
-  else
-    xmax = 1 - MA / E0;
+  G4double xmax = muonMass > MA ? muonMass / E0 : MA / E0;
```
Comment (hahnjo) [inline SimG4Core/CustomPhysics/src/G4muDarkBremsstrahlungModel.cc:266]: This changes the value of `xmax`, doesn't it? Before it was
```c++
  if ((muonMass / E0) > (MA / E0))
    xmax = 1 - muonMass / E0;
  else
    xmax = 1 - MA / E0;
```
and now it's equivalent to
```c++
  if ((muonMass / E0) > (MA / E0))
    xmax = muonMass / E0;
  else
    xmax = MA / E0;
```

**Comment by perrotta** [comment]
please test

**Comment by civanch** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**